### ISSUE TITLE
Why does binding an empty string resolve to null?

### ISSUE THREAD

**Issue by chasemyaccord** [NancyFx/Nancy#1996]
So I'm trying to bind some properties from a Javascript object to a model using Nancy and one of the properties is being changed from an empty string to null.  I narrowed it down to a method in the DefaultBinder class.

```
private bool BindingValueIsValid(string bindingValue, object existingValue, BindingMemberInfo modelProperty, BindingContext bindingContext)
{
    return (!String.IsNullOrEmpty(bindingValue) &&
            (IsDefaultValue(existingValue, modelProperty.PropertyType) ||
             bindingContext.Configuration.Overwrite));
}
```

I thought there may be a way to override this behavior with the BindingConfiguration but based on the above code, that wouldn't work anyway.  Is this a bug?  Or is there a reason why you'd always want to convert an empty string to null?   

Any input is appreciated.  Thanks!

**Comment by albertjan** [comment]
It's an oversight I think. It rarely happens that an empty string represents a value. I can't think of a way why this would be on purpose. 

You could do a pull request :smile:

**Comment by wtilton** [comment]
Pull request: https://github.com/NancyFx/Nancy/pull/2001

**Comment by thecodejunkie** [comment]
Moving this to 1.4 to be on par with #2001

**Comment by thecodejunkie** [comment]
Closed in conjunction with #2001
